includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -209,6 +208,33 @@ public function get_client() {
 		$this->google_client->setScopes( $this->get_required_scopes() );
 		$this->google_client->prepareScopes();
 
+		// This is called when the client refreshes the access token on-the-fly.
+		$this->google_client->setTokenCallback(
+			function( $cache_key, $access_token ) {
+				// These are sane defaults, since unfortunately they aren't passed.
+				$expires_in = HOUR_IN_SECONDS;
+				$created    = 0;
```

Comment:
Shouldn't `$created` be the current timestamp?

Reply:
The `set_access_token()` method takes care of that if `$created` is `0`, so no need to run that logic here, especially since it might be overridden below.